src/js/player.js

```diff
@@ -1225,6 +1288,30 @@ class Player extends Component {
    * @private
    */
   handleTechSourceset_(event) {
+    // only update the source cache when the source
+    // was not updated using the player api
+    if (!this.changingSrc_) {
```

Comment:
we could exist early if `this.changingSrc_` instead.

Reply:
yeah that looks nicer

Reply:
actually that will cause us to have to duplicate the `trigger` that happens below this block

Reply:
oh, I see, missed that piece.